DOCSP-39849: revise job batching docs (#2994)
jenssegers/laravel-mongodb: docs/queues.txt

   :values: tutorial

.. meta::
   :keywords: php framework, odm, code example, jobs

To use MongoDB as your database for Laravel Queue, change
the driver in your application's ``config/queue.php`` file:

.. code-block:: php

   'connections' => [
       'database' => [
           'driver' => 'mongodb',
           // You can also specify your jobs-specific database created on config/database.php
           'connection' => 'mongodb',

>            'collection' => 'jobs',
>            'queue' => 'default',
>            'retry_after' => 60,

Reviewer: In https://github.com/mongodb/laravel-mongodb/pull/2985#discussion_r1622733967 we decided with @jmikola to comment optional settings.
```suggestion
           // 'collection' => 'jobs',
           // 'queue' => 'default',
           // 'retry_after' => 60,
```

   :values: tutorial

.. meta::
   :keywords: php framework, odm, code example, jobs

To use MongoDB as your database for Laravel Queue, change
the driver in your application's ``config/queue.php`` file:

.. code-block:: php

   'connections' => [
       'database' => [
           'driver' => 'mongodb',
           // You can also specify your jobs-specific database
           // in the config/database.php file
           'connection' => 'mongodb',
           // 'collection' => 'jobs',
           // 'queue' => 'default',

>            // 'retry_after' => 60,

Reviewer 2: S: It's a little unclear why these fields are commented out - are they optional? I think either leaving them uncommented and highlighting line 24 or deleting them altogether would be better

Reviewer 2: Based on the table below, `collection` and `queue` are required so maybe leaving uncommented and highlighting line 24 is the way to go

     - Description

   * - ``driver``

>      - **Required**. Specifies the queue driver to use. The value of

Reviewer: I find it cumbersome to repeat "Specifies" for every setting.

     - Description

   * - ``driver``
     - **Required** Queue driver to use. The value of
       this property must be ``mongodb``.

   * - ``connection``
     - Database connection used to store jobs. It must be
       a ``mongodb`` connection. The driver uses the default connection
       if a connection is not specified.

   * - ``collection``
     - Name of the MongoDB collection to store failed
       jobs. The value is ``failed_jobs`` by default.

Then, add the service provider in your application's
``config/app.php`` file:

.. code-block:: php

   MongoDB\Laravel\MongoDBQueueServiceProvider::class,

Job Batching
------------

**Job batching** is a Laravel feature that enables you to execute a

> batch of jobs and subsequent actions before, after, and during the
> execution of the jobs from the queue. To learn more about this feature,

Reviewer 2: S: I think the use of "subsequent" is a little misleading, since these actions can also run before or during the execution of the jobs
```suggestion
batch of jobs and related actions before, after, and during the
execution of the jobs from the queue. To learn more about this feature,
```

see `Job Batching <https://laravel.com/docs/{+laravel-docs-version+}/queues#job-batching>`__
in the Laravel documentation.

In MongoDB, you don't have to create a designated collection before
using job batching. The ``job_batches`` collection is created

> automatically to store meta-information about your job batches, such as

Reviewer 2: S: metadata seems like the more familiar term

```suggestion
automatically to store metadata about your job batches, such as
```

     - Description

   * - ``driver``
     - **Required**. Specifies the queue driver to use. The value of
       this property must be ``mongodb``.

   * - ``connection``
     - Specifies the database connection used to store jobs. It must be a
       ``mongodb`` connection. The driver uses the default connection if
       a connection is not specified.

   * - ``collection``
     - Specifies the name of the MongoDB collection to store job
       batches. The value is ``job_batches`` by default.

> The, add the service provider in ``config/app.php``:

Reviewer: ```suggestion
Then, add the service provider in ``config/app.php``:
```

     - Description

   * - ``driver``
     - **Required** Queue driver to use. The value of
       this property must be ``mongodb``.

   * - ``connection``
     - Database connection used to store jobs. It must be a
       ``mongodb`` connection. The driver uses the default connection if
       a connection is not specified.

   * - ``collection``
     - Name of the MongoDB collection to store job
       batches. The value is ``job_batches`` by default.

> Then, add the service provider in ``config/app.php``:

Reviewer 2: S: since you introduced the other files this way

```suggestion
Then, add the service provider in your application's ``config/app.php`` file:
```
